src/server/main.py

```diff
@@ -31,23 +31,24 @@ def transform_taxon_results(taxa, ns, ns_config):
         taxon['ns'] = ns
 
 
-def transform_params(params, required_fields, field_name_remappings=None):
+def transform_query_params(params, required_ns_fields, field_name_remappings={}):
     """
     Set some defaults in the params that we pass into an RE query.
     `field_name_remappings` can be a dict of {'input_param_field_name': 'field_name_to_send_to_query'}
+    Cleans `search_text` for FULLTEXT search query reasons
+
     Mutates the params dict
     Returns the namespace name and the namespace config dict as a pair (see _NS_CONFIG below)
     """
-    if field_name_remappings:
-        for (input_name, output_name) in field_name_remappings.items():
-            params[output_name] = params[input_name]
-            del params[input_name]
-    params['ts'] = params.get('ts', int(time.time() * 1000))
-    ns = params['ns']
-    del params['ns']
+    ns = params.pop('ns')
     ns_config = _NS_CONFIG[ns]
-    for field_name in required_fields:
+    for field_name in required_ns_fields:
         params[field_name] = ns_config['query_params'][field_name]
```

Comment:
* Made use of `d.pop()` and `d.setdefault`
* `'ts'` was consistently being added in every calling function anyway
* Moved name remappings to end
* "Clean" search text by returning `''` when AQL would choke on it